vpr/src/route/rr_graph.cpp

```diff
@@ -2501,6 +2501,73 @@ static vtr::NdMatrix<std::vector<int>, 4> alloc_and_load_track_to_pin_lookup(vtr
     return track_to_pin_lookup;
 }
 
+/* Prints all the data about node inode to file fp.                    */
+void print_rr_node(FILE * fp, const std::vector<t_rr_node> &L_rr_node, int inode) {
+
+    std::string info = describe_rr_node(inode);
+    fprintf(fp, "%s\n", info.c_str());
+
+    fprintf(fp, "%d edge(s):", L_rr_node[inode].num_edges());
+    for (int iconn = 0; iconn < L_rr_node[inode].num_edges(); ++iconn)
+        fprintf(fp, " %d", L_rr_node[inode].edge_sink_node(iconn));
+    fprintf(fp, "\n");
+
+    fprintf(fp, "Switch types:");
+    for (int iconn = 0; iconn < L_rr_node[inode].num_edges(); ++iconn)
+        fprintf(fp, " %d", L_rr_node[inode].edge_switch(iconn));
+    fprintf(fp, "\n");
+
+    fprintf(fp, "Capacity: %d\n", L_rr_node[inode].capacity());
+    fprintf(fp, "R: %g  C: %g\n", L_rr_node[inode].R(), L_rr_node[inode].C());
+    fprintf(fp, "Cost_index: %d\n", L_rr_node[inode].cost_index());
+}
+
+/* Writes out al data about node inode to binary in file fp */
+void write_rr_node(FILE * fp, const std::vector<t_rr_node> &L_rr_node, int inode) {
```

Comment:
This doesn't write any of the edge metadata?  So this XML isn't suitable for use with ``genfasm``.

Reply:
So far that is correct.  We haven't focused much on generating fasm yet so this could potentially be a fairly easy fix.

Reply:
This kind of bespoke binary format is extremely fragile.  What other binary serializations have you considered?

Reply:
We haven't really considered any other serializations, this was just the fasted thing we would think of for prototyping because I have more C experience than C++.  If you have any suggestions for C++ style serialization I would definitely look into switching over to that when I have time.